chrome/content_scripts/content_annotation_handler.js

```diff
@@ -11,407 +11,256 @@
- * @return {void} - Note: the annotation is done inline.
- */
-function rubyWrap(node, regexs, consolidated_mapping) {
-	// Setting textContent on a node removes all of its children and replaces
-	// them with a single text node. Since we don't want to alter the DOM aside
-	// from annotating text, we only annotate on single text nodes.
-	// @see https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent
-	if (node.nodeType === Node.TEXT_NODE) {
-		// This node only contains text.
-		// @see https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType.
+		// observer.observe(document.body, {
+			// childList: true,
+			// subtree: true
+		// });
 
-		// Skip textarea nodes due to the potential for accidental annotation
-		// where none was intended.
-		if (node.parentNode && node.parentNode.nodeName === 'TEXTAREA') {
-			return;
+		/**
+		 * For efficiency, create a word --> search RegEx Map too.
+		 * @param {Object} map - An object with every key from each mapping selected for
+		 * 	annotation and a corresponding annotation
+		 * @returns {Map<String, RegExp>} A map of the keys from the full mapping onto
+		 * 	a case-insensitive RegExp for each key
+		 */
+		function createRegexMap(map) {
+			let regexs = new Map();
+			for (let key of Object.keys(map)) {
+				// We want a global, case-insensitive replacement.
+				// @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+				regexs.set(key, new RegExp(key, 'gi'));
+			}
+			return regexs;
 		}
 
-		// Because DOM manipulation is slow, we don't want to keep setting
-		// textContent after every replacement. Instead, manipulate a copy of
-		// this string outside of the DOM and then perform the manipulation
-		// once, at the end.
-		let content = node.textContent;
-		let match_found = false;
-
-		// Annotate every occurrence of 'word' in 'content' with its mapped value.
-		for (const [key, annotation] of Object.entries(consolidated_mapping)) {
-			// Grab the search regex for this word.
-			const regex = regexs.get(key);
-
-			// Actually do the wrapping.
-			// Note: if 'word' does not appear in 'content', nothing happens.
-			let key_match = [...content.matchAll(regex)][0];
-			if (key_match) {
-				match_found = true;
-				let alreadyWrapped = isInsideRubyTag(content, key_match);
-				if (!alreadyWrapped) {
-					let annotation_tag = {
-						key: key,
-						annotation: annotation
-					};
-					content = content.replace(
-						regex,
-						(match) => {
-							return `<ruby data-ie-annotated='${JSON.stringify(annotation_tag)}' style="font-size: 1.5em;">${key}<rp>(</rp><rt style="font-size: 1em">${annotation}&nbsp;</rt><rp>)</rp></ruby>`;
-						}
-					);
-				}
+		function messageListener(message) {
+			if (message.intended_recipient !== "RECIPIENT_CONTENT") {
+				return; // don't send response so intended recipient can still pick up this message
```

Comment:
Why is this `return;` where as the one in the background script is `return Promise.resolve()`?